frontend/dashboard/pages/CreateService/CreateService.test.tsx

```diff
@@ -191,6 +191,62 @@ describe('CreateService', () => {
     expect(emptyFieldErrors.length).toBe(1);
   });
 
+  it('should display loading while the form is processing', async () => {
+    const user = userEvent.setup();
+
+    renderWithMockServices(
+      {
+        // Use setTimeout as a workaround to trigger isLoading on mutation.
+        addRepo: () =>
+          new Promise((resolve) =>
+            setTimeout(() => {
+              resolve(repositoryMock);
+            }, 2),
+          ),
```

Comment:
Somehow `Promise.resolve` works perfectly with `useQuery`, but when I'm using it with `useMutation` `isLoading` is always `false`. 🤔 By mocking delay with 2ms with setTimeout the `isLoading` has time to be `true`. Do you have any clue why?

Reply:
I got this answer from chatGpt:  

When you use Promise.resolve, the promise is immediately resolved, and there might not be enough time for isLoading to be set to true before it's set to false. This could be due to the synchronous nature of Promise.resolve, where the resolution occurs in the same microtask.

When you use Promise.resolve, the resolution happens quickly in the same microtask, and there's not enough time for the isLoading flag to be set to true before it's set to false.

Reply:
Did you figure this out? I tried asking chatGPT, and tested som suggestions

Reply:
I did not. Did chatGTP give you some great suggestion?

Reply:
I wrote a comment here but i did not find it :innocent:
anyway this is what ChatGpt says:

When you use Promise.resolve directly, it resolves synchronously, so isLoading may not have time to become true before the state is checked. On the other hand, when you introduce a delay with setTimeout, it gives the useMutation hook enough time to update the loading state before it's checked.

Reply:
Promise.resolve is resolving synchronously (in microseconds), it means that the promise is being resolved immediately in the same microtask.

Reply:
Great! That's was my assumption too, hence I added the setTimeout method to delay the promise resolving with 2ms. 🙄😊 I'm not sure if that is the best solution or not though 😅